code/datums/records.dm

```diff
@@ -10,7 +10,7 @@
 	if(!copied)
 		copied = new src.type()
 	for(var/variable in src.vars)
-		if(variable in (SSrecords.excluded_fields || src.excluded_fields)) continue
+		if(variable in (SSrecords.excluded_fields | src.excluded_fields)) continue
```

Comment:
You're creating and destroying a temporary list for every iteration of this loop.